corehq/tests/test_locks.py

```diff
@@ -0,0 +1,66 @@
+from testil import assert_raises, eq
+
+from dimagi.utils.couch import get_redis_lock
+
+from corehq.util.test_utils import timelimit
+
+from .locks import ReentrantTestLock, reentrant_redis_locks
+from .noseplugins.redislocks import TestLock, TimeoutError
+
+
+def test_redislocks_nose_plugin():
+    lock1 = get_redis_lock(__name__, timeout=0.5, name="test")
+    assert isinstance(lock1.lock, TestLock), lock1.lock
+    lock1.acquire()
+    lock2 = get_redis_lock(__name__, timeout=0.5, name="test")
+    with assert_raises(TimeoutError):
+        lock2.acquire()
+    # lock1.release() -> LockError: Cannot release a lock that's no longer owned
```

Comment:
is this line meant to be a form of documentation of some kind? if the intention is that `lock1.release()` should throw a `LockError` could that not be part of the test?

Reply:
Yes, it's documentation to show why the lock, which is acquired above, is not released in this test (because it gets implicitly released on timeout). I'll change it to use `assert_raises` in a follow-up PR.

Reply:
https://github.com/dimagi/commcare-hq/pull/27402/commits/5e95fc2f85cfdf7602b8c8a442262d4b1f5285fc